js/@types/flarum/common/models/User.d.ts

```diff
@@ -0,0 +1,26 @@
+import _User from '../../../../../vendor/flarum/core/js/dist-typings/common/models/User';
+
+/**
+ * @inheritdoc
+ */
+export default class User extends _User {
```

Comment:
How will this scale if we have many extensions extending User?

Reply:
I haven't thought of any reasonable way to share the typings from one extension into another yet.

At the moment, things like this will only apply to this extension, so extensions will only see the things which they add to models like User.

Reply:
Fair enough, I don't see a much better solution either.

I wonder if we can export these so that if, say, an extension extends tags, it could use the tags-provided modifications of User and Discussion?

Reply:
sorry my typescript isn't that great, so excuse what might be a silly question, but isn't there a way we could extend typings ? something like this perhaps ?

https://www.damirscorner.com/blog/posts/20200911-ExtendingTypescriptTypeDefinitions.html#:~:text=A%20TypeScript%20type%20definition%20can,of%20the%20vue%20NPM%20package).

(though tsconfig wouldn't include extension typings anyway..)